arangod/Agency/AgencyPaths.h

```diff
@@ -638,6 +657,237 @@ class Root : public std::enable_shared_from_this<Root>, public Path {
       std::shared_ptr<Databases const> databases() const {
         return Databases::make_shared(shared_from_this());
       }
+
+      class ReplicatedLogs : public StaticComponent<ReplicatedLogs, Plan> {
+       public:
+        constexpr char const* component() const noexcept {
+          return "ReplicatedLogs";
+        }
+
+        using BaseType::StaticComponent;
+
+        class Database : public DynamicComponent<Database, ReplicatedLogs, DatabaseID> {
+         public:
+          char const* component() const noexcept { return value().c_str(); }
+
+          using BaseType::DynamicComponent;
+
+          class Log : public DynamicComponent<Log, Database, std::string> {  // TODO Use a different type than std::string?
```

Comment:
In some other places (even below in this same file) there is a type `DatabaseID` used for databases. I guess this is only a typedef for `std::string`, but it could also be used here and the TODO could be removed.

Reply:
But this does not refer to a DatabaseId but instead to a stringifyed LogId. Afaik dynamic components can only handle string values.